### PR TITLE
Raise psr/simple-cache to ^2.0|^3.0 and PHP to ^8.0

### PR DESCRIPTION
**Pull request by dehbka**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | [43](https://github.com/yiisoft/cache-file/issues/43)

### CODE REVIEW

**Review by rustamwin — COMMENTED** [review]
Version 7.4 needs to be removed from GitHub workflows. And of course need changelog line.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Yii FileCache Change Log
 
+## 2.0.0 July 10, 2022
```
Comment (rustamwin) [inline CHANGELOG.md:3]:
```suggestion
## 2.0.0 under development
```
The release date will be automatically generated when it is released.

**Review by rustamwin — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Yii FileCache Change Log
 
+## 2.0.0 under development
+
+- Chg: Raise the minimum `psr/simple-cache` version to `^2.0|^3.0` and the minimum PHP version to `^8.0`
```
Comment (rustamwin) [inline CHANGELOG.md:5]:
```suggestion
- Chg #44: Raise the minimum `psr/simple-cache` version to `^2.0|^3.0` and the minimum PHP version to `^8.0` (@dehbka)
```

**Review by rustamwin — COMMENTED** [review]
[These](https://github.com/yiisoft/cache-file/pull/44/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edL4-L8) lines should be removed.

**Review by rustamwin — APPROVED** [review]

File: src/FileCache.php [hunk 3]
```diff
@@ -424,9 +420,9 @@ private function gc(): void
     /**
      * @param mixed $key
```
Comment (vjik) [inline src/FileCache.php:421]: This is string. phpdoc can be removed

**Review by vjik — CHANGES_REQUESTED** [review]

File: src/FileCache.php [hunk 4]
```diff
@@ -365,9 +362,8 @@ private function getCacheFile(string $key): string
         $base = $this->cachePath;
 
         for ($i = 0; $i < $this->directoryLevel; ++$i) {
-            if (($prefix = substr($key, $i + $i, 2)) !== false) {
-                $base .= DIRECTORY_SEPARATOR . $prefix;
-            }
```
Comment (vjik) [inline src/FileCache.php:370]: From PHP 8 we should check on empty string instead false. Seems, need test for this case (in this PR or create issue).

**Review by vjik — CHANGES_REQUESTED** [review]

**Review by vjik — APPROVED** [review]

**Comment by samdark** [comment]
Thanks!

**Pull request merged**